test/language/expressions/class/fields-asi-4.js

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2017 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: ASI test in field declarations -- error when generator interpreted as multiplication
+esid: sec-automatic-semicolon-insertion
+features: [class-fields]
+---*/
+
+var C = class {
+  x = 42
+  *gen() {}
+}
```

Comment:
I believe this is an early parse error. you should use the negative flags instead.